crates/goose-cli/src/session/mod.rs

```diff
@@ -121,30 +118,33 @@ pub async fn classify_planner_response(
     }
 }
 
-impl Session {
+impl CliSession {
     pub fn new(
         agent: Agent,
-        session_file: Option<PathBuf>,
+        session_id: Option<String>,
         debug: bool,
         scheduled_job_id: Option<String>,
         max_turns: Option<u32>,
         edit_mode: Option<EditMode>,
         retry_config: Option<RetryConfig>,
     ) -> Self {
-        let messages = if let Some(session_file) = &session_file {
-            session::read_messages(session_file).unwrap_or_else(|e| {
-                eprintln!("Warning: Failed to load message history: {}", e);
-                Conversation::new_unvalidated(Vec::new())
+        let messages = if let Some(session_id) = &session_id {
+            tokio::task::block_in_place(|| {
+                tokio::runtime::Handle::current().block_on(async {
+                    SessionManager::get_session(session_id, true)
+                        .await
+                        .map(|session| session.conversation.unwrap_or_default())
+                        .unwrap()
+                })
```

Comment:
Hi, @DOsinga! Since this got merged, I started seeing this error: 

```
thread 'main' panicked at crates/goose-cli/src/session/mod.rs:136:26:
called `Result::unwrap()` on an `Err` value: Session not found
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Not sure if a good patch, but this fixed the issue for me locally: https://github.com/block/goose/pull/5055